libbeat/autodiscover/providers/docker/config.go

```diff
@@ -36,15 +40,24 @@ type Config struct {
 
 func defaultConfig() *Config {
 	return &Config{
-		Host:   "unix:///var/run/docker.sock",
-		Prefix: "co.elastic",
+		Host:      "unix:///var/run/docker.sock",
+		Prefix:    "co.elastic",
+		Separator: "/",
 	}
 }
 
 // Validate ensures correctness of config
-func (c *Config) Validate() {
+func (c *Config) Validate() error {
 	// Make sure that prefix doesn't ends with a '.'
 	if c.Prefix[len(c.Prefix)-1] == '.' && c.Prefix != "." {
 		c.Prefix = c.Prefix[:len(c.Prefix)-2]
 	}
+	if len(c.Separator) != 1 {
+		return fmt.Errorf("separator '%v' must be a single character", c.Separator)
+	}
+	// Make sure that separator isn't included in prefix
+	if strings.Contains(c.Prefix, c.Separator) {
```

Comment:
Notice that `/` is a safe separator because it is not accepted in docker or kubernetes containers, but `-` or `.` would produce ambiguities as they can be used in container names.